File: codeforces/714_B.py
n = int(input())
arr = [*map(int, input().split())]
arr = set(arr)
 
if len(arr) > 3:
    print("NO")
if len(arr) < 3:
    print("YES")
if len(arr) == 3:
    arr = list(arr)
    arr.sort()
    if abs(arr[1] - arr[0]) == abs(arr[2] - arr[1]):
        print("YES")
    else:
        print("NO")

File: codeforces/714_B.cpp
#include <bits/stdc++.h>

using ll = long long;
using namespace std;

int main(){
    int n; cin >> n;
    int els[n];
    for(int &el: els) cin >> el;
    sort(els, els + n);
    bool good = true;
    if(n <= 3){

    }
    if(good) cout << "YES";
    else cout << "NO";
    return 0;
}